packages/jellyfish-api-core/src/category/wallet.ts

```diff
@@ -412,3 +423,43 @@ export interface WalletFlagResult {
   flag_state: boolean
   warnings: string
 }
+
+export interface InWalletTransaction {
+  amount: number
+  fee: number
+  confirmations: number
+  blockhash: string
+  blockindex: number
+  blocktime: number
+  txid: string
+  time: number
+  timereceived: number
+  bip125replaceable: BIP125
+  details: InWalletTransactionDetail[]
+  hex: string
+}
+
+interface InWalletTransactionDetail {
+  address: string
+  category: InWalletTransactionCategory
+  amount: number
+  label: string
+  vout: number
+  fee: number
+  abandoned: boolean
+}
```

Comment:
some of these conditions are different to set in your transaction, don't need to do them all but do basic things tests for amount, address, label, vout, confirmations, blockindex .e.g

Reply:
I realized I could set up the transaction using `sendToAddress` instead of using a magic value for the `txid` however, the data that comes back is missing keys like `blockhash`, `blockindex`. Does this mean that these are actually optional properties? I couldn't quite figure out from the `gettransaction` function in `rpcwallet.cpp`

Reply:
This means they are not yet included in a block. For it to be mined, you need to `container.generate(1)`.